lib/mongo/uri.rb

```diff
@@ -615,6 +626,94 @@ def auth_mech_props(value)
       properties
     end
 
+    # Parses the zlib compression level.
+    #
+    # @param value [ String ] The zlib compression level string.
+    #
+    # @return [ Integer | nil ] The compression level value if it is between -1 and 9 (inclusive),
+    #   otherwise nil (and a warning will be raised).
+    def zlib_compression_level(value)
+      if /^-?\d+$/ =~ value
+        i = value.to_i
+
+        if i >= -1 && i <= 9
+          return i
+        end
+      end
+
+      log_warn("#{value} is not a valid zlibCompressionLevel")
+      nil
+    end
+
+
+    # Parses a boolean value.
+    #
+    # @param value [ String ] The URI option value.
+    #
+    # @return [ true | false | nil ] The boolean value parsed out, otherwise nil (and a warning
+    #   will be raised).
+    def bool(value)
+      case value
+      when "true"
+        true
+      when "false"
+        false
+      else
+        log_warn("invalid boolean: #{value}")
```

Comment:
This log message should also specify what the option was that the value goes with, to assist in troubleshooting.

Reply:
Fixed